components/simple-paypal.jsx

```diff
@@ -1,44 +1,68 @@
 var React = require('react');
+import { FormattedHTMLMessage, IntlMixin } from 'react-intl';
+
+var PaypalInput = React.createClass({
+  render: function() {
+    return (
+      <input
+        onChange={this.props.donateOnChange}
+        onKeyDown={this.props.donateKeyDown}
+        type="number" name="donation_amount_other" min={this.props.minAmount} placeholder={this.props.placeholder} className="medium-label-size" required value={this.props.value}/>
+    );
+  }
+});
+
+var PaypalButton = React.createClass({
+  render: function() {
+    var message = this.props.defaultMessage;
+    if (this.props.value) {
+      message = this.props.message
+                  .replace("{currency_symbol}", this.props.currencySymbol)
+                  .replace("{amount}", this.props.value);
+    }
+    return (
+      <button type="submit" className="btn large-label-size" id="donate-btn">
+        {message}
+      </button>
+    );
+  }
+});
 
 var simplePaypal = React.createClass({
+  mixins: [IntlMixin],
+  getInitialState: function() {
+    var value = "";
+
+    return {
+      value: ""
+    };
+  },
+  donateOnChange: function(event) {
+    this.setState({
+      value: event.target.value
+    });
+  },
+  donateKeyDown: function(event) {
+    var functionKeys = [46, 8, 9, 27, 13, 110, 190, 37, 39]; // backspace, delete, tab, escape, enter . left, and right
+    var numberKeys = [ 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 96, 97, 98, 99, 100, 101, 102, 103, 104, 105 ]; // numbers
+    var allowed = functionKeys.concat(numberKeys);
+    if (allowed.indexOf(event.keyCode) === -1) {
+      event.preventDefault();
+    }
+  },
   componentDidMount: function() {
-    var currencySymbol = this.props.currencySymbol;
     // ***********************************************
     // Update Donate button to make it show the selected donation amount
     // ***********************************************
     var $theForm = $('#primary-form');
 
-    $theForm.find('[name="donation_amount_other"]').val("");
-
-    function updateDonateButtonText(amountSelected) {
-      var buttonText = amountSelected ? "Donate " + currencySymbol + "{ amount } now".replace("{ amount }", amountSelected) : "Donate now";
-      $("#donate-btn").text(buttonText);
-      $('#paypal-one-time').find('[name="amount"]').attr('value', amountSelected);
-      $('#paypal-recurring').find('[name="a3"]').attr('value', amountSelected);
-    }
-
-    function updateDonateButtonTextEvent(event) {
-      var amountSelected = $(this).val();
-      updateDonateButtonText(amountSelected);
-    }
-
     // check if a pre-selected amount is passed via the URL
     if (window.location.hash.match(/#amount-\d+?/)) {
-      var preSelected = window.location.hash.substring(8);
-      $theForm.find('[name="donation_amount_other"]').val(preSelected);
-      updateDonateButtonText(preSelected);
+      this.setState({
+        value: window.location.hash.substring(8)
```

Comment:
I tried and wasn't super happy with it.

It requires more work.

```
import Router from 'react-router';
console.log(Router.HashLocation.getCurrentPath())
```
vs
```
console.log(window.location.hash);
```

Not only that, but `Router.HashLocation.getCurrentPath` doesn't work server side, it fails on getting `window.location.hash`...

Feels super redundant to me.